### PR TITLE
gh-83004: Clean up refleak in _zoneinfo initialisation

### PR DESCRIPTION
**Pull request by hauntsaninja**
* Issue: gh-83004

### CODE REVIEW

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Code looks good, but I think we need a NEWS entry as this fixes a (very unlikely) user-facing issue.

**Comment by hauntsaninja** [comment]
Added the NEWS entry. If you want to review other changes like this, there's https://github.com/python/cpython/pull/98840 and https://github.com/python/cpython/pull/98841

I also went through CPython and about 20 other modules are affected. Should I continue to do single module PRs or can I batch them?

**Comment by JelleZijlstra** [comment]
Thanks! I think batching them is fine since the change is simple.

File: Misc/NEWS.d/next/Library/2022-11-02-05-54-02.gh-issue-83004.0v8iyw.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Clean up refleak in :mod:`_zoneinfo` initialisation
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Library/2022-11-02-05-54-02.gh-issue-83004.0v8iyw.rst]:
```suggestion
Clean up refleak on failed module initialisation in :mod:`_zoneinfo`
```
Clarify this is an unlikely edge case

**Review by JelleZijlstra — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**